### PR TITLE
Add screen for editing local safe name

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #750 

Changes proposed in this pull request:
- Add screen for editing local safe name
- Adjust nav graph
- React to local name changes in ui
     - Updating main toolbar
     - Updating name item

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsEditNameFragment.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            backButton.setOnClickListener {
+                close()
+            }
+            safeName.showKeyboardForView()
+            safeName.setOnEditorActionListener listener@ { v, actionId, event ->
+                if (actionId == EditorInfo.IME_ACTION_DONE) {
+                    viewModel.saveLocalName(safeName.text.toString())
+                    return@listener true
```
Comment (rmeissner) [inline app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsEditNameFragment.kt:45]: So we only save the name if you press done on the keyboard? If you close the keyboard and go back the name will not be saved, right. Not sure if this is super intuitive, since there is no explicit save button. I would check this with @posthnikova and @tschubotz

File: app/src/test/java/io/gnosis/safe/ui/settings/safe/SafeSettingsEditNameViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    @get:Rule
+    val instantExecutorRule = TestLifecycleRule()
+
+    private val safeRepository = mockk<SafeRepository>(relaxed = true)
+    private lateinit var viewModel: SafeSettingsEditNameViewModel
+
+    @Test
+    fun `saveLocalName (name) - should change local name for safe`() = runBlockingTest {
+
```
Comment (rmeissner) [inline app/src/test/java/io/gnosis/safe/ui/settings/safe/SafeSettingsEditNameViewModelTest.kt:31]: This only tests the happy case. Would love to see test for some edge cases (e.g. no safe or error when interacting with the repo). But not super important right now.
